rmf_site_editor/src/widgets/fuel_asset_browser.rs

```diff
@@ -16,24 +16,37 @@
 */
 
 use crate::{
-    interaction::{ModelPreviewCamera, ObjectPlacement, PlaceableObject, Selection},
-    site::{
-        AssetSource, CurrentLevel, FuelClient, Model, ModelLoader, SetFuelApiKey, UpdateFuelCache,
-    },
+    interaction::{ModelPreviewCamera, ObjectPlacementExt},
+    site::{AssetSource, FuelClient, Model, ModelLoader, SetFuelApiKey, UpdateFuelCache},
     widgets::prelude::*,
-    AppState, CurrentWorkspace,
 };
 use bevy::{ecs::system::SystemParam, prelude::*};
 use bevy_egui::egui::{self, Button, ComboBox, ImageSource, RichText, ScrollArea, Ui, Window};
 use gz_fuel::FuelModel;
 
 /// Add a [`FuelAssetBrowser`] widget to your application.
-#[derive(Default)]
-pub struct FuelAssetBrowserPlugin {}
+pub struct FuelAssetBrowserPlugin {
+    pub model_spawner: fn(&mut Commands, Model),
+}
+
+impl Default for FuelAssetBrowserPlugin {
+    fn default() -> Self {
+        Self {
+            model_spawner: |commands: &mut Commands, model: Model| {
+                commands.place_object_2d(model);
+            },
+        }
+    }
+}
+
+/// Stores the function that uses commands to place models for this application
+#[derive(Resource, Deref, DerefMut)]
+struct ModelPlacer(fn(&mut Commands, Model));
```

Comment:
I have no problem using a function pointer for this, but in the future we could consider making this a `BoxedSystem` for added extensibility. The downside would be a lot of extra management of borrows, so we can definitely punt on that until we find a situation where it matters.